src/repository/opamRepository.ml

```diff
@@ -80,9 +80,17 @@ let fetch_from_cache =
         ~quiet:true ~validate:false ~overwrite:true ~checksum
         url file
     | `rsync ->
-      (OpamLocal.rsync_file url file @@| function
-        | Result _ | Up_to_date _-> ()
-        | Not_available (s,l) -> raise (OpamDownload.Download_fail (s,l)))
+      begin match OpamUrl.local_file url with
+        | Some src ->
+          let src = OpamFilename.to_string src in
+          let file = OpamFilename.to_string file in
+          OpamSystem.copy_file src file;
```

Comment:
```suggestion
          OpamFilename.copy_file src file;
```
Better use `OpamFilename`, than `OpamSystem` (it's a redirection)

Reply:
Done. Thanks a lot, I wasn't aware this function existed.